src/lib/client.js

```diff
@@ -259,13 +257,19 @@ class Client extends EventEmitter {
     reqMessage.data.id = uuid()
     return new Promise((resolve, reject) => {
       const done = (err, res) => {
-        clearTimeout(timeout)
-        // Wait till nextTick to remove the listener so that it doesn't happen while the
-        // event is part way through being emitted, which causes issues iterating the listeners.
-        process.nextTick(() => {
-          this.plugin.removeListener('incoming_message', responseListener)
-        })
-        return err ? reject(err) : resolve(res)
+        if (err) return reject(err)
+
+        if (res.data.method === 'quote_response') {
```

Comment:
`_sendAndReceiveMessage` shouldn't refer to quotes; it's more generic than that. The reference to `quote_response` should be in `_getQuote`.

Also, this introduces a bug: if there's an error, the timeout isn't cleared and the `incoming_message` listener is never removed.